atlp-rwanda/hackers-ec-be
# #187419070 A buyer should be able to add and remove a product to wish list

#### What does this PR do?
Allow buyers to add or remove products from their wishlist.

#### Description of Task to be completed?
Make the following endpoints functional:
- POST /api/v1/wishes
- GET /api/v1/wishes
- GET /api/v1/wishes/id

#### How should this be manually tested?
To test these changes:

1. Clone the repository and go to its directory.
2. Install dependencies by running npm install.
3. Create a .env file based on env-example, filling in the required variables.
4. Start the app with npm run dev.
5. To add or remove products from the wishlist, use the POST method at "api/v1/wishes", passing the buyer's token and product ID.
6. To view the wishlist for a buyer or a seller, use the GET method at "api/v1/wishes", passing the token. Sellers see products and the number of users interested, while buyers see their wished products.
7. Use the GET method at "api/v1/wishes/id" to view whether a specific product is wished or not.

#### Relevant Pivotal Tracker/Trello Stories
This PR corresponds to Pivotal Tracker story #187419070.
